apps/web/src/services/dashboards/shared/dashboard-widget-input-form/helpers/reference-handler-helper.ts

```diff
@@ -30,10 +32,17 @@ const getResources = async (inputText: string, options: ReferenceHandlerOptions)
         const resourceType = schemaProperty.reference?.resource_type;
         const distinctKey = schemaProperty.reference?.reference_key;
 
+        // NOTE: Some variables(asset) require specific API filters.
+        if (schemaProperty.title === ASSET_VARIABLE_TYPE_INFO.asset_query_set.name) {
+            filterHelper.setFilters([{ k: 'ref_cloud_service_type.labels', o: '=', v: 'Compliance' }]);
+        } else if (schemaProperty.title === ASSET_VARIABLE_TYPE_INFO.asset_account.name) {
+            filterHelper.setFilters([{ k: 'provider', o: '=', v: 'aws' }]);
+        }
+
         if (filters?.length) {
             const resourceKey = distinctKey ?? REFERENCE_TYPE_INFO[propertyName]?.key;
             if (resourceKey) {
-                filterHelper.setFilters([{ k: resourceKey, v: filters.map((d) => d.name as string), o: '=' }]);
+                filterHelper.addFilter({ k: resourceKey, v: filters.map((d) => d.name as string), o: '=' });
```

Comment:
issue: I think there is possibility to reuse previous filters. please initiate `filterHelper` with `filterHelper.setFilters([])`.